File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int get_letters(string text);
int get_words(string text);
int get_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    int letters = get_letters(text);
    int words = get_words(text);
    int sentences = get_sentences(text);

    float l = (float)letters / words * 100;
    float s = (float)sentences / words * 100;

    int index = round(0.0588 * l - 0.296 * s - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int get_letters(string text)
{
    int letters = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {

        if (isalpha(text[i]))
        {
            letters++;
        }
    }

    return letters;
}

int get_words(string text)
{
    int words = 1;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }

    return words;
}

int get_sentences(string text)
{
    int sentences = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        char c = text[i];
        if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }

    return sentences;
}

File: week-6_python/pset/readability.py
def main():
    text = input("Text: ")

    letters = letters_count(text)
    words = words_count(text)
    sentences = sentences_count(text)

    l = (letters / words) * 100
    s = (sentences / words) * 100

    index = round(0.0588 * l - 0.296 * s - 15.8)

    if (index < 1):
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")

def letters_count(text):
    count = 0
    for char in text:
        if char.isalpha():
            count += 1
    return count

def words_count(text):
    words = 1
    for char in text:
        if char.isspace():
            words += 1
    return words

def sentences_count(text):
    sentences = 0
    for char in text:
        if char == '.' or char == '!' or char == '?':
            sentences += 1
    return sentences

if __name__ == "__main__":
    main()